contracts/LpSugar.vy

```diff
@@ -389,13 +512,113 @@ def _byData(_data: address[3], _account: address) -> Lp:
     emissions: emissions,
     emissions_token: emissions_token,
 
-    account_balance: pool.balanceOf(_account),
-    account_earned: earned,
-    account_staked: acc_staked,
-
-    pool_fee: pool_fee,
+    staked_fee: pool_fee,
+    unstaked_fee: pool_fee,
     token0_fees: token0.balanceOf(pool_fees),
-    token1_fees: token1.balanceOf(pool_fees)
+    token1_fees: token1.balanceOf(pool_fees),
+
+    positions: positions
+  })
+
+@internal
+@view
+def _byDataCL(_data: address[3], _account: address) -> Lp:
+  """
+  @notice Returns CL pool data based on the factory, pool and gauge addresses
+  @param _data The addresses to lookup
+  @param _account The user account
+  @return Lp struct
+  """
+  pool: IV3Pool = IV3Pool(_data[1])
+  gauge: ICLGauge = ICLGauge(_data[2])
+  nft: INFTPositionManager = INFTPositionManager(pool.nft())
+
+  gauge_fees: GaugeFees = pool.gaugeFees()
+  gauge_alive: bool = self.voter.isAlive(gauge.address)
+  fee_voting_reward: address = empty(address)
+  emissions: uint256 = 0
+  emissions_token: address = empty(address)
+  token0: IERC20 = IERC20(pool.token0())
+  token1: IERC20 = IERC20(pool.token1())
+
+  if gauge.address != empty(address):
+    fee_voting_reward = gauge.feesVotingReward()
+    emissions_token = gauge.rewardToken()
+
+  if gauge_alive:
+    emissions = gauge.rewardRate()
+
+  slot: Slot = pool.slot0()
+  # https://blog.uniswap.org/uniswap-v3-math-primer
+  sqrt_price: uint160 = slot.sqrt_price / (2**96)
+  # Do we want to normalize for token decimals here?
+  price: uint160 = sqrt_price**2
```

Comment:
I'm thinking maybe we should just leave it as `price: uint160 = pool.slot0().sqrt_price` and we deal with it in the front-end directly?